samples/api/hlsl_shaders/hlsl_shaders.cpp

```diff
@@ -0,0 +1,513 @@
+	        VK_SHADER_STAGE_FRAGMENT_BIT,
+	        1)};
+	descriptor_layout_create_info =
+	    vkb::initializers::descriptor_set_layout_create_info(
+	        set_layout_bindings.data(),
+	        static_cast<uint32_t>(set_layout_bindings.size()));
+	VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout_create_info, nullptr, &base_descriptor_set_layout));
+
+	// Set layout for the samplers
+	set_layout_bindings = {
+	    // Binding 0: Fragment shader sampler
+	    vkb::initializers::descriptor_set_layout_binding(
+	        VK_DESCRIPTOR_TYPE_SAMPLER,
+	        VK_SHADER_STAGE_FRAGMENT_BIT,
+	        0)};
+	descriptor_layout_create_info =
+	    vkb::initializers::descriptor_set_layout_create_info(
+	        set_layout_bindings.data(),
+	        static_cast<uint32_t>(set_layout_bindings.size()));
+	VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout_create_info, nullptr, &sampler_descriptor_set_layout));
+
+	// Pipeline layout
+	// Set layout for the base descriptors in set 0 and set layout for the sampler descriptors in set 1
+	std::vector<VkDescriptorSetLayout> set_layouts = {base_descriptor_set_layout, sampler_descriptor_set_layout};
+	VkPipelineLayoutCreateInfo         pipeline_layout_create_info =
+	    vkb::initializers::pipeline_layout_create_info(
+	        set_layouts.data(),
+	        static_cast<uint32_t>(set_layouts.size()));
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &pipeline_layout));
+}
+
+void HlslShaders::setup_descriptor_set()
+{
+	// We separate the descriptor sets for the uniform buffer + image and samplers, so we don't need to duplicate the descriptors for the former
+	VkDescriptorSetAllocateInfo descriptor_set_alloc_info{};
+
+	// Descriptors set for the uniform buffer and the image
+	descriptor_set_alloc_info =
+	    vkb::initializers::descriptor_set_allocate_info(
+	        descriptor_pool,
+	        &base_descriptor_set_layout,
+	        1);
+	VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &descriptor_set_alloc_info, &base_descriptor_set));
+
+	VkDescriptorBufferInfo buffer_descriptor = create_descriptor(*uniform_buffer_vs);
+
+	// Combined image descriptor for the texture
+	VkDescriptorImageInfo image_descriptor{};
+	image_descriptor.imageView   = texture.image->get_vk_image_view().get_handle();
+	image_descriptor.imageLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+	image_descriptor.sampler     = texture.sampler;
+
+	std::vector<VkWriteDescriptorSet> write_descriptor_sets = {
+	    // Binding 0 : Vertex shader uniform buffer
+	    vkb::initializers::write_descriptor_set(
+	        base_descriptor_set,
+	        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+	        0,
+	        &buffer_descriptor),
+	    // Binding 1 : Fragment shader sampled image
+	    vkb::initializers::write_descriptor_set(
+	        base_descriptor_set,
+	        VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
```

Comment:
The descriptor set layout has this as VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE not COMBINED. This causes a validation issue:

[error] [framework/core/instance.cpp:41] 1165064310 - VUID-VkGraphicsPipelineCreateInfo-layout-00756: Validation Error: [ VUID-VkGraphicsPipelineCreateInfo-layout-00756 ] Object 0: handle = 0x55cfd1c74f90, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x45717876 | Type mismatch on descriptor slot 0.1 (expected `VK_DESCRIPTOR_TYPE_SAMPLER, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER`) but descriptor of type VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE The Vulkan spec states: layout must be consistent with all shaders specified in pStages (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkGraphicsPipelineCreateInfo-layout-00756)

Reply:
Good catch. Somehow didn't notice that one.